src/plugins/data/common/expressions/interpreter_provider.ts

```diff
@@ -52,6 +52,14 @@ export function interpreterProvider(config: any) {
   async function invokeChain(chainArr: any, context: any): Promise<any> {
     if (!chainArr.length) return Promise.resolve(context);
 
+    // if execution was aborted return error
```

Comment:
I moved this up here because if the signal has already been aborted, no need to execute the next function in the chain before returning an error.